pageserver/src/tenant/filename.rs

```diff
@@ -172,6 +172,118 @@ impl fmt::Display for ImageFileName {
         )
     }
 }
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum LayerFileName {
+    Image(ImageFileName),
+    Delta(DeltaFileName),
+    #[cfg(test)]
+    Test(String),
+}
+
+impl LayerFileName {
+    pub fn file_name(&self) -> String {
+        match self {
+            LayerFileName::Image(fname) => format!("{fname}"),
+            LayerFileName::Delta(fname) => format!("{fname}"),
+            #[cfg(test)]
+            LayerFileName::Test(fname) => fname.to_string(),
+        }
+    }
+    #[cfg(test)]
+    pub(crate) fn new_test(name: &str) -> LayerFileName {
+        LayerFileName::Test(name.to_owned())
+    }
+}
+
+impl From<ImageFileName> for LayerFileName {
+    fn from(fname: ImageFileName) -> Self {
+        LayerFileName::Image(fname)
+    }
+}
+impl From<DeltaFileName> for LayerFileName {
+    fn from(fname: DeltaFileName) -> Self {
+        LayerFileName::Delta(fname)
+    }
+}
+
+// include a `/` in the name as an additional layer of robustness
+// because `/` chars are not allowed in UNIX paths
+#[cfg(test)]
+pub const LAYER_FILE_NAME_TEST_PREFIX: &str = "LAYER_FILE_NAME::test/";
+
+impl<'a> TryFrom<&'a str> for LayerFileName {
+    type Error = String;
+
+    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+        #[cfg(test)]
+        if let Some(value) = value.strip_prefix(LAYER_FILE_NAME_TEST_PREFIX) {
+            return Ok(LayerFileName::Test(value.to_owned()));
+        }
+        let delta = DeltaFileName::parse_str(value);
+        let image = ImageFileName::parse_str(value);
+        let ok = match (delta, image) {
+            (None, None) => {
+                return Err(format!(
+                    "neither delta nor image layer file name: {value:?}"
+                ))
+            }
+            (Some(delta), None) => LayerFileName::Delta(delta),
+            (None, Some(image)) => LayerFileName::Image(image),
+            (Some(_), Some(_)) => unreachable!(),
+        };
+        Ok(ok)
+    }
+}
+
+impl LayerFileName {
+    pub fn display(&self) -> impl std::fmt::Display {
```

Comment:
I think that should be removed and only `file_name` should be used, if anybody wants to print it.
I've checked in the code, and it looks that we now log some of such file names, for example:

> info!("scheduled layer file deletion {}", name.display());

which will be turned into something 

`scheduled layer file deletion 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__00000000016B59D8-00000000016B5A51`

It's not clear from the layer name, where does it gets deleted from.
Some places contain spans with IDs, but at least one (mentioned in the other comment below in `timeline.rs`) seems to miss them.
I think we should check the other places after deletion, and use the full file path where applicable.

Reply:
Done.

I think I checked all the places.

Full path is only appropriate to use

1. during Timeline's get page code where we cross the ancestor boundary
2. in error messages that involve reading from the layer file.

In all other places, we have the tenant_id and timeline_id in tracing scope.

Did appropriate changes, will push them soon.